plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/CachingDirectoryLister.java

```diff
@@ -144,21 +146,25 @@ private static RemoteIterator<TrinoFileStatus> createListingRemoteIterator(Trino
     }
 
     @Override
-    public void invalidate(Location location)
+    public void invalidate(Location location, SchemaTableName schemaTableName)
     {
-        cache.invalidate(location);
+        log.debug("Invalidating cache for schemaTableName: %s and location: %s", schemaTableName, location);
+        cache.invalidate(new CacheKey(location, schemaTableName));
     }
 
     @Override
     public void invalidate(Table table)
     {
         if (isCacheEnabledFor(table.getSchemaTableName()) && isLocationPresent(table.getStorage())) {
             if (table.getPartitionColumns().isEmpty()) {
-                cache.invalidate(Location.of(table.getStorage().getLocation()));
+                log.debug("Invalidating (table) cache for table: %s", table.getSchemaTableName());
+                cache.invalidate(new CacheKey(Location.of(table.getStorage().getLocation()), table.getSchemaTableName()));
             }
             else {
                 // a partitioned table can have multiple paths in cache
-                cache.invalidateAll();
+                SchemaTableName tableName = table.getSchemaTableName();
+                log.debug("Invalidating (partitions) cache for table: %s", table.getSchemaTableName());
```

Comment:
I find it confusing that i see sometimes (partitions) and sometimes (table). Just leave them out. It is cache.

Reply:
Maybe:
- `log.debug("Invalidating cache for partitioned table: %s", table.getSchemaTableName());`
- and `log.debug("Invalidating cache for unpartitioned table: %s", table.getSchemaTableName());` ?

Reply:
i can just remove logging, no worries.  would that make things easier?

Reply:
this is a  nit comment - no need to overfocus on it :)